docs/man/git-lfs-config.5.ronn

```diff
@@ -223,6 +223,15 @@ lfs option can be scoped inside the configuration for a remote.
   * `total` The entire size of the file, in bytes.
   * `name` The name of the file.
 
+* `GIT_LFS_SET_LOCKABLE_READONLY`
+  `lfs.setlockablereadonly`
+
+  These settings, the first an environment variable and the second a gitconfig
+  setting, control whether files marked as 'lockable' in `git lfs track` are
+  made read-only in the working copy when not locked by the current user.
+  The default is `true`; you can disable this behaviour and have all files
+  writeable by setting either variable to 0, 'no' or 'false'.
```

Comment:
@ttaylorr If the lockable feature is on, then Git LFS cannot [skip](https://github.com/git-lfs/git-lfs/blob/a6a5ba2f38d364f60d4f1b4b5361888fa76575bd/commands/command_post_checkout.go#L29) the `post-checkout` command. Consequently, Git LFS needs to [search](https://github.com/git-lfs/git-lfs/blob/a6a5ba2f38d364f60d4f1b4b5361888fa76575bd/git/attribs.go#L149-L159) for all `.gitattributes` files to check them for lockable patterns.

My team is working with a large repo (>20k directories, >300k files) and just searching for the `.gitattributes` file in all directories takes ~6 sec on macOS with a fast SSD. Consider these timings:


```
$ export GIT_LFS_SET_LOCKABLE_READONLY=1; time git-lfs post-checkout a0c5174a19f2dece44c09bc76f91feb69799e66f a0c5174a19f2dece44c09bc76f91feb69799e66f 0
git-lfs post-checkout a0c5174a19f2dece44c09bc76f91feb69799e66f  0  

35.09s user 6.41s system 730% cpu 5.679 total


$ export GIT_LFS_SET_LOCKABLE_READONLY=0; time git-lfs post-checkout a0c5174a19f2dece44c09bc76f91feb69799e66f a0c5174a19f2dece44c09bc76f91feb69799e66f 0
git-lfs post-checkout a0c5174a19f2dece44c09bc76f91feb69799e66f  0  

0.01s user 0.02s system 89% cpu 0.030 total
```

I plan to distribute the setting `lfs.setlockablereadonly=false` in my company gitconfig but it took me a while to figure out the culprit of the slowness. I don't know how widely the lockable feature is used, but I wonder if `lfs.setlockablereadonly=false`is a more sensible default? Alternatively we could find ways to speed up the `.gitattributes` discovery.

Reply:
> I don't know how widely the lockable feature is used, but I wonder if `lfs.setlockablereadonly=false` is a more sensible default?

I think `=true` is an important default value to keep. Here are a few thoughts on how we can get the best of both worlds:

1. Speed up the time it takes to search for `.gitattributes`. This has been optimized by @sinbad in https://github.com/git-lfs/git-lfs/pull/1616. I think that this work could be augmented by first `stat()`-ing directories to search, and then seeing if the stat is out-of-date with the last time that the directory was traversed, effectively memoizing the search. I presume that this would improve the speed a great deal.

2. Warn when the search is taking too long. We can set a timer (similar to what Microsoft does with GVFS when the `--ahead-behind` check takes too long) and warn if Git LFS is unable to determine the set of lockable files quickly.

What do you think? Perhaps we should have this discussion in a new issue, as well.